packages/service/db/migrations/20210726224018_role-auth.sql

```diff
@@ -0,0 +1,21 @@
+-- migrate:up
+ALTER TABLE document ENABLE ROW LEVEL SECURITY;
+
+CREATE FUNCTION current_user_id() RETURNS uuid AS $$
+  SELECT nullif(current_setting('app.user_id', true), '')::uuid;
+$$ LANGUAGE SQL STABLE;
+
+CREATE ROLE postgraphile_user;
+
+CREATE POLICY select_if_creator ON document
+  FOR SELECT
+  TO postgraphile_user
+  USING (creator_id = current_user_id());
```

Comment:
This isn't exactly what I want. What this does, is allow SELECT for any row where the `current_user_id` is the `creator_id`, but I want to allow SELECTs for:
```sql
current_user_id() IN (
  SELECT document_user.user_id 
  FROM document_user WHERE document_user.document_id = document.id
)
```

also